### PR TITLE
22331 - adding body to the filing items

### PR DESCRIPTION
**Pull request by hfekete**
*Issue:22331 *https://github.com/bcgov/entity/issues/22331 

*Description of changes:*
- adding body to the filings
- adding common components
- cleaning up and refactoring some of the code (extracting to components, reorganizing js code)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-20-h92oph9t.web.app

File: src/components/bcros/businessDetails/Links.vue [hunk 1]
```diff
@@ -145,8 +146,8 @@ const promptChangeBusinessInfo = () => {
   const baseUrl = useRuntimeConfig().public.editApiURL
   const editUrl = `${baseUrl}/${currentBusiness.value.identifier}`
 
-  // if (!isGoodStanding && !isRoleStaff) {
-  if (!currentBusiness.value.goodStanding) {
+  // if (!isGoodStanding && !hasRoleStaff) {
```
Comment (kialj876) [inline src/components/bcros/businessDetails/Links.vue:149]: rmv comment?

File: src/components/bcros/filing/common/HeaderActions.vue [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    case isFilingType(props.filing, undefined, FilingSubTypeE.FULL_RESTORATION):
+      return true // not supported
+    case isFilingType(props.filing, FilingTypes.SPECIAL_RESOLUTION):
+      return true // not supported
+    case isFilingType(props.filing, FilingTypes.TRANSITION):
+      return true // not supported
+  }
+
+  // eslint-disable-next-line no-console
+  console.log('disableCorrection(), unhandled filing =', props.filing)
```
Comment (kialj876) [inline src/components/bcros/filing/common/HeaderActions.vue:155]: can we make this an 'info' instead and you can remove the eslint-disable?

**Review by kialj876 — COMMENTED** [review]
Really nice work! Just needs some tests. Also, if its easy can you add loader spin when you click to download a document?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-20-h92oph9t.web.app

**Review by kialj876 — APPROVED** [review]
Nice work

**Pull request merged**